### PR TITLE
Update Docker installation 

### PR DESCRIPTION
**Pull request by Graeme-Miller**
Update Docker installation to use new Docker apt-get repositories. This will allow us to use docker 1.8 and 1.9

### CODE REVIEW

File: docker/src/main/java/brooklyn/entity/container/docker/DockerHostSshDriver.java [hunk 1]
```diff
@@ -289,13 +289,25 @@ public String getVersion() {
...
 
     private String installDockerOnUbuntu() {
-        String version = getVersion();
-        log.debug("Installing Docker version {} on Ubuntu", version);
+        final String dockerVersion = getVersion();
+        final String ubuntuVersion = getMachine().getMachineDetails().getOsDetails().getVersion();
+
+        final String dockerRepoName;
+        final String repositoryVersionName;
+        switch(ubuntuVersion){
```
Comment (grkvlt) [inline docker/src/main/java/brooklyn/entity/container/docker/DockerHostSshDriver.java]: Add spaces: `switch (ubuntuVersion) {`

File: docker/src/main/java/brooklyn/entity/container/docker/DockerHostSshDriver.java [hunk 2]
```diff
@@ -289,13 +289,25 @@ public String getVersion() {
...
+            case "12.04": dockerRepoName = "ubuntu-precise"; repositoryVersionName = dockerVersion+ "-0~precise"; break;
```
Comment (grkvlt) [inline docker/src/main/java/brooklyn/entity/container/docker/DockerHostSshDriver.java]: split all of these onto multiple lines:
```Java
case "12.04":
    dockerRepoName = "ubuntu-precise"; repositoryVersionName = dockerVersion+ "-0~precise";
    break;
```

File: docker/src/main/java/brooklyn/entity/container/docker/DockerHostSshDriver.java [hunk 3]
```diff
@@ -289,13 +289,25 @@ public String getVersion() {
     }
 
     private String installDockerOnUbuntu() {
-        String version = getVersion();
-        log.debug("Installing Docker version {} on Ubuntu", version);
+        final String dockerVersion = getVersion();
```
Comment (grkvlt) [inline docker/src/main/java/brooklyn/entity/container/docker/DockerHostSshDriver.java]: Not sure why any of these are `final` so suggest removing it from all

File: docker/src/main/java/brooklyn/entity/container/docker/DockerHostSshDriver.java [hunk 4]
```diff
@@ -289,13 +289,25 @@ public String getVersion() {
...
+        final String ubuntuVersion = getMachine().getMachineDetails().getOsDetails().getVersion();
+
```
Comment (grkvlt) [inline docker/src/main/java/brooklyn/entity/container/docker/DockerHostSshDriver.java]: Remove this blank line but add one before the `switch` statement

File: docker/src/main/java/brooklyn/entity/container/docker/DockerHostSshDriver.java (hunk 3 above)
Comment (Graeme-Miller) [inline docker/src/main/java/brooklyn/entity/container/docker/DockerHostSshDriver.java]: It's good practice
"In general, declare all variables final if they are not modified after being initialized or set to some value."
https://docs.oracle.com/cd/E19159-01/819-3681/abebf/index.html

File: docker/src/main/java/brooklyn/entity/container/docker/DockerHostSshDriver.java (hunk 4 above)
Comment (Graeme-Miller) [inline docker/src/main/java/brooklyn/entity/container/docker/DockerHostSshDriver.java]: According to Oracle's programming style, this is correct:
"One blank line should always be used in the following circumstances:
Between the local variables in a method and its first statement
..
Between logical sections inside a method to improve readability"
http://www.oracle.com/technetwork/java/javase/documentation/codeconventions-141388.html

**Pull request merged**